src/protocol/sort/apply_sort/apply_sort_permutation.rs

```diff
@@ -0,0 +1,116 @@
+use crate::{
+    error::Error,
+    ff::Field,
+    protocol::{
+        context::{Context, SemiHonestContext},
+        sort::{apply::apply_inv, generate_permutation::shuffle_and_reveal_permutation},
+    },
+    secret_sharing::Replicated,
+};
+
+use super::shuffle_objects::{shuffle_object_shares, Resharable};
+use crate::protocol::sort::ApplyInvStep::ShuffleInputs;
+use crate::protocol::sort::SortStep::ShuffleRevealPermutation;
+
+#[derive(Debug)]
+pub struct SortPermutation<F: Field>(pub Vec<Replicated<F>>);
+
+impl<F: Field> SortPermutation<F> {
+    #[allow(dead_code)]
+    /// ## Panics
+    /// It will propagate panics from sort
+    /// # Errors
+    /// it will propagate errors from sort
+    pub async fn apply_sort_permutation<I>(
```

Comment:
```suggestion
    pub async fn apply<I>(
```

... might be enough given that you can call this "sort_permutation"

Reply:
Makes sense.. Getting rid of "permutation" also, just calling the file apply_sort